src/pages-and-resources/discussions/data/api.js

```diff
@@ -0,0 +1,130 @@
+import { getConfig } from '@edx/frontend-platform';
+import { getAuthenticatedHttpClient } from '@edx/frontend-platform/auth';
+
+function normalizeLtiConfig(data) {
+  if (!data || Object.keys(data).length < 1) {
+    return {};
+  }
+
+  return {
+    consumerKey: data.lti_1p1_client_key,
+    consumerSecret: data.lti_1p1_client_secret,
+    launchUrl: data.lti_1p1_launch_url,
+  };
+}
+
+function normalizePluginConfig(data) {
+  if (!data || Object.keys(data).length < 1) {
+    return {};
+  }
+
+  return {
+    allowAnonymousPosts: data.allow_anonymous,
+    allowAnonymousPostsPeers: data.allow_anonymous_to_peers,
+    blackoutDates: JSON.stringify(data.discussion_blackouts),
+    // TODO: We need all these added to the API.  ... default them for now
+    divideByCohorts: false,
+    allowDivisionByUnit: false,
+    divideCourseWideTopics: false,
+    // TODO: Note that these last two are in the `discussion_topics` data, but we haven't been able
+    // to add them properly here yet.  I'm not sure the data is in a usable state as is, since it
+    // only seems to include the topics for which these would be "true".  Assuming its only these
+    // two, I think we could check for the existence of "General" for the first, but I'm not sure
+    // what the topic title is for the second.  "Questions for TAs" maybe?
+    divideGeneralTopic: false,
+    divideQuestionsForTAs: false,
+  };
+}
+
+function normalizeApps(data) {
+  const apps = Object.entries(data.providers.available).map(([key, app]) => ({
+    id: key,
+    featureIds: app.features,
+    // TODO: Fix this and get it from the backend!
+    documentationUrl: 'http://example.com',
+    hasFullSupport: app.features.length >= data.features.length,
+  }));
+  return {
+    courseId: data.context_key,
+    enabled: data.enabled,
+    features: data.features.map(id => ({
+      id,
+    })),
+    appConfig: {
+      id: data.providers.active,
+      ...normalizePluginConfig(data.plugin_configuration),
+      ...normalizeLtiConfig(data.lti_configuration),
+    },
+    activeAppId: data.providers.active,
+    apps,
+  };
+}
+
+function denormalizeData(courseId, appId, data) {
+  /*
+  TODO: What about these?  Some are from the instructor dashboard, presumably... but I don't think they all are.
+
+  divideByCohorts
+  allowDivisionByUnit
+  divideCourseWideTopics
+  divideGeneralTopic
+  divideQuestionsForTAs
```

Comment:
I don't think equivalents for all these exist yet. 
AFAIK current you can get the discussion settings from `/courses/{courseid}/discussions/settings` however it is sturctured differently (for once you can divide by enrolment tracks as well) and there is no concept of divide questions for TAs.

Reply:
Good to know, thanks - I was concerned about the Questions for TAs in particular.  This jives with what we're seeing in the backend data model as well.